test/random-battles/gen9.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Tests for Gen 9 randomized formats
+ */
+'use strict';
+
+const {testTeam} = require('./tools');
+const assert = require('../assert');
+const Teams = require('./../../sim/teams').Teams;
+const Dex = require('./../../sim/dex').Dex;
+
+describe('[Gen 9] Random Battle', () => {
+	const options = {format: 'gen9randombattle'};
+	const setsJSON = require(`../../data/random-sets.json`);
+	const dex = Dex.forFormat(options.format);
+
+	it('all Pokemon should have 4 moves, except for Ditto (slow)', function () {
+		// This test takes more than 2000ms
+		testTeam({...options, rounds: 100}, team => {
+			for (const pokemon of team) assert(pokemon.name === 'Ditto' || pokemon.moves.length === 4);
+		});
+	});
+
+	it('all moves on all sets should be obtainable (slow)', function () {
+		const generator = Teams.getGenerator(options.format);
+		const rounds = 100;
+		for (const pokemon of Object.keys(setsJSON)) {
+			const species = dex.species.get(pokemon);
+			const sets = setsJSON[pokemon]["sets"];
+			const types = species.types;
+			const abilities = new Set(Object.values(species.abilities));
+			if (species.unreleasedHidden) abilities.delete(species.abilities.H);
+			for (const set of sets) {
+				const role = set.role;
+				const movePool = set.movepool.map(m => dex.moves.get(m).id);
+				const teraTypes = set.teraTypes;
+				// Every move in the movePool should be obtainable
+				for (const move of movePool) {
```

Comment:
I wonder if it would be more efficient to create a `Set` out of the movepool, remove the moves when they are found, and then break once the set is empty.

Reply:
> I wonder if it would be more efficient to create a `Set` out of the movepool, remove the moves when they are found, and then break once the set is empty.

Good idea, I will try that

Reply:
Seems to work :)